### PR TITLE
feat: add a button to share logs

### PR DESCRIPTION
**Pull request by 1010Tom**
Share the logs as file (Even though the logs are not useful yet)

IMG_81A54E326C29-1

### CODE REVIEW

File: mobile/lib/common/settings_screen.dart [hunk 1]
```diff
@@ -66,6 +73,19 @@ class SettingsScreen extends StatelessWidget {
...
+        ElevatedButton(
+            onPressed: () async {
+              var file = await FLog.exportLogs();
+              var logsAsString = await file.readAsString();
+              final List<int> bytes = utf8.encode(logsAsString);
+              final Directory tempDir = await getTemporaryDirectory();
+              String now = DateFormat('yyyy-MM-dd_HH:mm:ss').format(DateTime.now());
+              final String filePath = '${tempDir.path}/$now.logs';
+              await File(filePath).writeAsBytes(bytes);
+              final XFile logFile = XFile(filePath);
```
Comment (holzeis) [inline mobile/lib/common/settings_screen.dart:85]: 🤔  `Flog.exportLogs()` is already writing the file to the disk, why do read it and write it again? Shouldn't it be suffcient to simply do the following?

```
final file = await.FLog.exportLogs();
find XFile logFile = XFile(file.path);
String now = DateFormat('yyyy-MM-dd_HH:mm:ss').format(DateTime.now());
Share.shareXFiles([logFile], text: 'Logs from $now');
```

**Review by holzeis — APPROVED** [review]

File: mobile/lib/common/settings_screen.dart (hunk 1 above)
Comment (bonomat) [reply mobile/lib/common/settings_screen.dart:85]: I tried this but it did not work.

**Comment by bonomat** [comment]
bors retry

**Pull request merged**